src/OMODFrameworkWrapper.cpp

```diff
@@ -164,9 +165,11 @@ OMODFrameworkWrapper::EInstallResult OMODFrameworkWrapper::install(MOBase::Guess
   try
   {
     MessageBoxHelper::unique_ptr messageBoxHelper = MessageBoxHelper::make_unique();
+    auto cphDeleter = [](CodeProgressHelper* cph) { cph->deleteLater(); };
+    std::unique_ptr<CodeProgressHelper, decltype(cphDeleter)> codeProgressHelper(new CodeProgressHelper(mParentWidget), cphDeleter);
```

Comment:
I think it's probably better to have the code progress helper owned by the CodeProgress object. The OMODFramework settings object is static, so there's nothing actually stopping us setting up the settings (including the CodeProgress object) in `InstallerOMOD::init` to save on duplicated work. It's only *not* doing that because I was basically copying form my first attempt, which used OMOD-Framework instead of OMODFramework, and didn't have static settings, and I haven't got around to tidying it up yet.

Without this change, all that needs doing is making `initFrameworkSettings` static and moving the call site, but with it it gets significantly more complicated. If the managed object owns the helper, though, it doesn't make it any more complicated.

Reply:
Also, the version here would leave a dangling pointer in the managed object between OMOD installs, which isn't ideal.

Reply:
I tried having `CodeProgressHelper` being owned by `CodeProgress` similar to `ScriptFunction`, but it makes the installation crash at the end... You can take a loot at it if you want.

Reply:
Don't do it like `ScriptFunction` - when that gets disposed (`~`), we want to fun the finaliser (`!`), but CodeProgress' disposer gets called every time the popup is hidden, so we can't do that. As the .NET object will probably stay around until the application quits, it's probably a good idea to connect something to one of Qt's everything's-being-closed signals so we don't call `deleteLater` on a QObject that's already been destroyed if .NET ends up being torn down after Qt.

Reply:
something like this:

```cpp
CodeProgress::~CodeProgress()
{
  // hide popup
}

CodeProgress::!CodeProgress()
{
  // implicitly checks a hidden member variable, so this will only ever run once
  if (mHelper)
    mHelper->deleteLater();
}

AnotherClass::AnotherMethod()
{
  connect(QApplication::instance(), &QApplication::aboutToQuit, [=](){mCodeProgress->mHelper->deleteLater(); mCodeProgress->mHelper = nullptr;});
}
```

Reply:
I think I'll handle this myself when things get split up. Annoyingly, the C++/CLI compiler inserts `System::GC::SuppressFinalizer(this)` into the disposer (`~`), so the finalizer (`!`) won't get called. Normally, this is useful as normally the same `IDisposable` doesn't get reused, so you'd call the finalizer from the disposer.